packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -265,15 +265,36 @@ export class SchemaFactory<TScope extends string = string, TName extends number
 		class schema extends this.nodeSchema(name, NodeKind.Object, t, true) {
 			public constructor(input: InsertableObjectFromSchemaRecord<T>) {
 				super(input);
+
+				// Differentiate between when this class is being constructor vs. a subclass
+				// of this class is being constructed:
+				//
+				// Case 1: Direct construction (POJO)
+				//
+				//     const Foo = _.object("Foo", {bar: _.number});
+				//     assert.deepEqual(new Foo({ bar: 42 }), { bar: 42 },
+				//		   "Prototype chain equivalent to POJO.");
+				//
+				// Case 2: Subclass construction (Domain Object)
+				//
+				// 	   class Foo extends _.object("Foo", {bar: _.number}) {}
+				// 	   assert.notDeepEqual(new Foo({ bar: 42 }), { bar: 42 },
+				// 	       "Subclass prototype chain differs from POJO.");
+				//
+				// In Case 1 (POJO), the prototype chain should be equivalent to '{}' (proxyTarget = undefined)
+				// In Case 2 (Domain Object), the prototype chain include the user's subclass (proxyTarget = this)
+				const customizable = this.constructor !== schema;
+				const proxyTarget = customizable ? this : undefined;
+
```

Comment:
It turns out that Craig already did something very similar for '_.array()' and '_.map()', which is why deepEquals() was working for a subset of tests (and why 'instanceof TreeNode' was unreliable with the pre-existing code.)